### PR TITLE
Replaced Naveen's implementation with sklearn's functions

### PR DESCRIPTION
**Pull request by sophieball**
Many changes:

1. Fixed some typos which should've surfaced earlier - they should've been called each time the code was executed.
2. Used sklearn's funtions to implement 10-fold nested cross validation, as described in the [paper](https://cmustrudel.github.io/papers/raman20toxicity.pdf). The results look similar #46 
Along the same line, I added back hyper-parameter tuning.
3. Added two more features, number of ? and number of !. They don't seem to help.
4. Using length or the number of words seems to mess up the model, which produces all 0's as labels. I think I had that as one of the features yesterday when I was getting all 0's in the output.
5. Instead of using ~3000 non-toxic and ~100 toxic comments, I fed in ~1000 non-toxic and ~100 toxic comments. It produced better results. The parameter tuning results suggests that when the ratio between non-toxic and toxic data is 10:1, the result is the best. I wonder if it's just because it's using all the data. In any case, I'll fetch more labeled data from MongoDB.
6. Removed some functions that are not used. Formatted code in files that I touched (`src/text_modifier.py`).

### CODE REVIEW

File: src/suite.py [hunk 1]
```diff
@@ -432,22 +362,22 @@ def remove_I(self, test_issues):
 
     return test_issues
 
-  def remove_SE(self, test_issues):
+  def remove_SE(self, data):
 
     features = self.features
     tf_idf_counter = self.tf_idf_counter
-    model = self.model
+    #model = self.model
```
Comment (CaptainEmerson) [inline src/suite.py:369]: rm

File: src/suite.py [hunk 2]
```diff
@@ -533,36 +450,78 @@ def all_combinations(self, function, matched_pairs=False):
...
-          average="weighted",
-          beta=0.5)
-
-      return score
-
-    best_score = self.all_combinations(
-        self_issue_classification_statistics_per, matched_pairs=matched_pairs)
-    logging.info("Best f0.5 score: {}".format(best_score))
-    return self.model
+    global model
```
Comment (CaptainEmerson) [inline src/suite.py:453]: global variable? are you sure you need it?

File: src/suite.py [hunk 3]
```diff
@@ -533,36 +450,78 @@ def all_combinations(self, function, matched_pairs=False):
...
+    logging.info("Best parameter: {}, ratio: {}.".format(
+        best_model.best_estimator_, best_ratio))
+    logging.info("Best score: {}.".format(best_model.best_score_))
+    self.model = best_model
+
+    #self.cross_validation()
+
+    #logging.info("Removing angry words towards oneself and SE words.")
+    y_train = self.all_train_data["label"]
+    X_train = self.all_train_data[self.features]
```
Comment (CaptainEmerson) [inline src/suite.py:499]: lowecase?

File: src/suite.py [hunk 4]
```diff
@@ -533,36 +450,78 @@ def all_combinations(self, function, matched_pairs=False):
...
+    logging.info("Trying all combinations of hyper parameters.")
+    logging.info("Scores with {}-fold cross validation".format(n_splits))
+    logging.info("Best parameter: {}, ratio: {}.".format(
+        best_model.best_estimator_, best_ratio))
+    logging.info("Best score: {}.".format(best_model.best_score_))
+    self.model = best_model
+
+    #self.cross_validation()
+
+    #logging.info("Removing angry words towards oneself and SE words.")
```
Comment (CaptainEmerson) [inline src/suite.py:497]: re-add?

File: src/suite.py [hunk 5]
```diff
@@ -533,36 +450,78 @@ def all_combinations(self, function, matched_pairs=False):
...
+
+    # Find the optimal parameters
+    logging.info("Trying all combinations of hyper parameters.")
+    logging.info("Scores with {}-fold cross validation".format(n_splits))
+    logging.info("Best parameter: {}, ratio: {}.".format(
+        best_model.best_estimator_, best_ratio))
+    logging.info("Best score: {}.".format(best_model.best_score_))
+    self.model = best_model
+
+    #self.cross_validation()
```
Comment (CaptainEmerson) [inline src/suite.py:495]: rm?

File: src/suite.py (hunk 3 above)
Comment (CaptainEmerson) [reply src/suite.py:499]: also you've got a commetned out log statement

File: src/suite.py (hunk 2 above)
Comment (sophieball) [reply src/suite.py:453]: It's always there.. You are right.. It's causing the confusion I discussed with you yesterday, about whether the model is updating itself with more data. I'll remove it.. in a new PR =p

> This is a big PR so it's hard to evaluate comprehensively. Putting your 6 changes into different PRs would make your reviewer's life easier. :)

Right.. adding features and changing data size can certainly be a separate PR.

File: src/suite.py (hunk 3 above)
Comment (sophieball) [reply src/suite.py:499]: > lowecase?

you mean X_train? I'd like to leave it as X_train, like in the tutorials I followed - if I needed to copy their code, it'd been easier.

File: src/suite.py (hunk 4 above)
Comment (sophieball) [reply src/suite.py:497]: Added.

File: src/suite.py (hunk 5 above)
Comment (sophieball) [reply src/suite.py:495]: Hmm.. I left it there because that is a easy switch between my sklearn version and Naveen's version.

**Comment by sophieball** [comment]
Ah.. not yet pushed the new code

File: src/suite.py (hunk 5 above)
Comment (sophieball) [reply src/suite.py:495]: Should I add an if-else to make it a switch?

Comment (sophieball) [reply src/suite.py:495]: I'll leave it like that for now.
